source/features/notifications-filter.tsx

```diff
@@ -0,0 +1,140 @@
+type Category = 'Type' | 'Status' | 'Read';
+
+async function handleSelection({target}: Event): Promise<void> {
+	const activeFilters: Record<Category, string[]> = {
+		Type: [],
+		Status: [],
+		Read: []
+	};
+	for (const selectedFilter of select.all('[aria-checked="true"]', target as Element)) {
+		activeFilters[selectedFilter.dataset.category as Category].push(filters[selectedFilter.dataset.filter as Filter]);
+	}
+
+	for (const notification of select.all('.notifications-list-item')) {
+		notification.hidden = false;
+		for (const [category, categoryFilters] of Object.entries(activeFilters)) {
+			if (categoryFilters.length === 0) {
+				continue;
+			}
+
+			if (category === 'Read') {
+				if (categoryFilters.length === 1 && !notification.classList.contains(categoryFilters[0])) {
+					notification.hidden = true;
+					break;
+				}
+
+				continue;
+			}
+
+			if (!select.exists(categoryFilters, notification)) {
+				notification.hidden = true;
+				break;
+			}
+		}
+	}
+
+	// Hide empty notifications groups
+	for (const group of select.all('.js-notifications-group')) {
+		group.hidden = !select.exists('.notifications-list-item:not([hidden])', group);
+	}
+}
+
+const icons: Record<Filter, JSX.Element> = {
+	PR: <GitPullRequestIcon className="color-text-secondary"/>,
+	Issue: <IssueOpenedIcon className="color-text-secondary"/>,
+	Open: <CheckCircleIcon className="color-text-success"/>,
+	Closed: <XCircleIcon className="color-text-danger"/>,
+	Draft: <GitPullRequestDraftIcon className="color-text-tertiary"/>,
+	Merged: <GitMergeIcon className="text-purple"/>,
+	Read: <DotIcon className="color-text-link"/>,
+	Unread: <DotFillIcon className="color-text-link"/>
+};
+
+function createDropdownList(category: Category, filters: Filter[]): JSX.Element {
+	return (
+		<div className="SelectMenu-list">
+			<header className="SelectMenu-header">
+				<span className="SelectMenu-title">{category}</span>
+			</header>
+			{filters.map(filter => (
+				<label
+					className="SelectMenu-item text-normal"
+					role="menuitemcheckbox"
+					aria-checked="false"
+					tabIndex={0}
+					data-filter={filter}
+					data-category={category}
+				>
+					<CheckIcon className="octicon octicon-check SelectMenu-icon SelectMenu-icon--check mr-2" aria-hidden="true"/>
+					<div className="SelectMenu-item-text">
+						<input hidden type="checkbox"/>
+						{icons[filter]}
+						<span className="ml-2">{filter}</span>
+					</div>
+				</label>
+			))}
+		</div>
+	);
+}
+
+function createDropdown(): JSX.Element {
+	return (
+		<details className="details-reset details-overlay position-relative d-none d-md-block">
+			<summary
+				className="btn ml-3"
+				data-hotkey="f"
+				aria-haspopup="menu"
+				role="button"
+			>
+				Filter by <span className="dropdown-caret ml-1"/>
+			</summary>
+			<details-menu
+				className="SelectMenu left-0"
+				aria-label="Filter by"
+				role="menu"
+				on-details-menu-selected={handleSelection}
```

Comment:
Using `on-details-menu-selected` instead of `on-details-menu-select` means we don't have to do `await delay(1)` in the event handler (this event is sent _after_ the dropdown DOM is updated).

I think this could be applied to `conversation-activity-filter` (untested).